### DISCUSSION TITLE
Usage of quarkus-maven-plugin in multi-module project

### DISCUSSION THREAD

**Discussion by bobanr** [quarkusio/quarkus#41684] [Q&A]
Hello. I'm new to Quarkus and quarkus-maven-plugin. 

Have the following:
- I'm having a multi-module quarkus project, for ex. with the following hierarchy:

parent
├── container
├── soap-api

- Have one container sub-module, that has the quarkus-maven-plugin and is basically the runnable module 
- Have another soap-api sub-module that uses quarkus-cxf for creating soap API. This sub-module also requires quarkus-maven-plugin in order to support the quarkis-cxf for generating code based on given wsdl. This should not be runnable. 
- The problems comes when trying to execute **quarkus dev** in the parent. Since having quarkus-maven-plugin in both container and soap-api sub-modules, it tries to run the soap-api as well. 
- Is there a way to prevent running the dev in the soap-api module, while executing the quarkus dev in the parent?

[a comment by an automated account is not shown]

**Reply by aloubyansky** [reply]
Does `soap-api` configure the `build` goal of the `quarkus-maven-plugin`? If it doesn't it shouldn't be launching it in dev mode.

Otherwise, you could simply run it from the `container` module. Or, I suppose, `mvn -f container quarkus:dev`.

**Reply by bobanr** [reply]
At the moment, this is how the plugin configured in the soap-api, so I don't have build goal. 
I also tried to disable default goals and explicitly just define generate-sources, but still, running quarkus dev on the parents, tries to run dev mode on the soap-api as well 

            <plugin>
                <groupId>io.quarkus.platform</groupId>
                <artifactId>quarkus-maven-plugin</artifactId>
                <extensions>true</extensions>
                <configuration>
                    <debug>false</debug>
                </configuration>
                <executions>
                    <execution>
                        <id>default</id>
                        <phase>none</phase>
                    </execution>
                    <execution>
                        <id>generate-sources</id>
                        <goals>
                            <goal>generate-code</goal>
                            <goal>generate-code-tests</goal>
                        </goals>
                    </execution>
                    <execution>
                        <id>default-cli</id>
                        <phase>none</phase>
                    </execution>
                </executions>
            </plugin>

**Reply by aloubyansky** [reply]
Do you have a quarkus-maven-plugin configuration in the root pom?

**Reply by bobanr** [reply]
Eventually, in some upper level parent pom (our parent has another parent, out of our project), the plugin is defined in pluginManagement as:

                <plugin>
                    <groupId>io.quarkus.platform</groupId>
                    <artifactId>quarkus-maven-plugin</artifactId>
                    <version>${version.micro.quarkus}</version>
                    <extensions>true</extensions>
                    <executions>
                        <execution>
                            <goals>
                                <goal>build</goal>
                                <goal>generate-code</goal>
                                <goal>generate-code-tests</goal>
                            </goals>
                        </execution>
                    </executions>
                </plugin>
                
But shouldn't our configuration override that one?

**Reply by aloubyansky** [reply]
By default the configuration is inherited and merged. You can try tricks described in https://maven.apache.org/pom.html#plugins
